converters/provider-aws/kafka/cluster.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Upbound Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	srcv1alpha1 "github.com/crossplane-contrib/provider-aws/apis/kafka/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/pkg/errors"
+	"github.com/upbound/extensions-migration/converters/common"
+	targetv1beta1 "github.com/upbound/provider-aws/apis/kafka/v1beta1"
+	"github.com/upbound/upjet/pkg/migration"
+)
+
+func ClusterResource(mg resource.Managed) ([]resource.Managed, error) {
+	source := mg.(*srcv1alpha1.Cluster)
+	target := &targetv1beta1.Cluster{}
+	if _, err := migration.CopyInto(source, target, targetv1beta1.Cluster_GroupVersionKind,
+		"spec.forProvider.configurationInfo", "spec.forProvider.brokerNodeGroupInfo", "spec.forProvider.clientAuthentication",
+		"spec.forProvider.encryptionInfo", "spec.forProvider.loggingInfo", "spec.forProvider.openMonitoring"); err != nil {
+		return nil, errors.Wrap(err, "failed to copy source into target")
+	}
+	target.Spec.ForProvider.Region = &source.Spec.ForProvider.Region
+	if source.Spec.ForProvider.CustomConfigurationInfo != nil {
+		target.Spec.ForProvider.ConfigurationInfo = make([]targetv1beta1.ConfigurationInfoParameters, 1)
+		target.Spec.ForProvider.ConfigurationInfo[0].Arn = source.Spec.ForProvider.CustomConfigurationInfo.ARN
+		target.Spec.ForProvider.ConfigurationInfo[0].Revision = common.PtrFloat64FromInt64(source.Spec.ForProvider.CustomConfigurationInfo.Revision)
+	}
+	if source.Spec.ForProvider.CustomBrokerNodeGroupInfo != nil {
```

Comment:
nit:
the following fields are newly introduced at the target API. might be good idea to document it for the consumers.
`.Spec.ForProvider.BrokerNodeGroupInfo[0].AzDistribution`
`.Spec.ForProvider.BrokerNodeGroupInfo[0].EBSVolumeSize`
also it started to accept xp reference and selector
`.Spec.ForProvider.EncryptionInfo[0].EncryptionAtRestKMSKeyArnRef`
`.Spec.ForProvider.EncryptionInfo[0].EncryptionAtRestKMSKeyArnSelector`

Reply:
Added documentation!